application/app/controllers/dataverse/datasets_controller.rb

```diff
@@ -107,4 +107,13 @@ def search_files_page
       return
     end
   end
+
+  def redirect_to_safe_referer(fallback: root_path, **options)
```

Comment:
not sure about the name. What about `redirect_back_to_app` or similar?
safe is a bit misleading.